include/neml2/tensors/functions/where.h

```diff
@@ -25,6 +25,7 @@
 #pragma once
 
 #include "neml2/tensors/tensors_fwd.h"
+#include "neml2/misc/types.h"
```

Comment:
Forward declaration of `TraceableTensorShape` should be enough.

Reply:
It is not.  Nor is forward declaring `ATensor`.